### PR TITLE
[4.0] Adding a manage button to the sample data module

### PR DESCRIPTION
**Pull request by Bakual**
As an partial alternative to #33165

### Summary of Changes
Adds a button to the sample data module which leads to the plugin manager. Sampledata plugins will be prefiltered.

This certainly needs improvement on the design aspect (where I suck badly). So if anyone has an idea how this can be improved I'm open to suggestions or PRs :-)

### Testing Instructions
Just use the button

### Actual result BEFORE applying this Pull Request
image

### Expected result AFTER applying this Pull Request
image

Buttons leads to
image

### Documentation Changes Required
None

### CODE REVIEW

File: administrator/modules/mod_sampledata/tmpl/default.php [hunk 1]
```diff
@@ -54,6 +54,12 @@
 			</li>
 		<?php endforeach; ?>
 	</ul>
+	<a href="index.php?option=com_plugins&filter[folder]=sampledata" class="float-end">
+		<button type="button" class="btn btn-secondary btn-sm manage-sample-data">
```
Comment (dgrammatiko) [inline administrator/modules/mod_sampledata/tmpl/default.php]: That's not correct, keep only the link tag and delete the button

**Comment by Quy** [comment]
Why not select `Unpublish` under the cog icon or add `Manage` to it if it is possible?

**Comment by Bakual** [comment]
> Why not select `Unpublish` under the cog icon or add `Manage` to it if it is possible?

Because that cog icon comes from the module chrome (not from the module itself). To my knowledge the module can't manipulate this part.

File: administrator/modules/mod_sampledata/tmpl/default.php (hunk 1 above)
Comment (Bakual) [reply administrator/modules/mod_sampledata/tmpl/default.php]: Thanks, removed the button.

**Comment by ChristineWk** [comment]
I have tested this item :white_check_mark: successfully on df52e896f75ba6547d27995c9ca6352eb53de5e2<br /><br />about the design: removing bottom border, as shown here #33106 and maybe .float-end margin? to hv alignment with the other (Install)<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33169">issues.joomla.org/tracker/joomla-cms/33169</a>.</sub>

**Comment by infograf768** [comment]
please solve conflicts as #33168 was merged

**Comment by infograf768** [comment]
I may have another solution to directly edit the plugin (color can be modified)

sampledata_plugins

**Comment by Bakual** [comment]
A button to directly edit (or unpublish) the sample data plugin isn't possible easy because we don't know the ID of the plugin.
That would require changes in each sample data plugins onSampledataGetOverview method.
It can certainly be done, but has much more impact.

**Comment by infograf768** [comment]
We need to fetch the db. here is the modified file
```
<?php
/**
 * @package     Joomla.Administrator
 * @subpackage  mod_sampledata
 *
 * @copyright   (C) 2017 Open Source Matters, Inc. <https://www.joomla.org>
 * @license     GNU General Public License version 2 or later; see LICENSE.txt
 */

defined('_JEXEC') or die;

use Joomla\CMS\Factory;
use Joomla\CMS\Language\Text;
use Joomla\CMS\Router\Route;
use Joomla\CMS\Session\Session;
use Joomla\CMS\Uri\Uri;

$app->getDocument()->getWebAssetManager()
	->registerAndUseScript('mod_sampledata', 'mod_sampledata/sampledata-process.js', [], ['defer' => true], ['core']);

Text::script('MOD_SAMPLEDATA_CONFIRM_START');
Text::script('MOD_SAMPLEDATA_ITEM_ALREADY_PROCESSED');
Text::script('MOD_SAMPLEDATA_INVALID_RESPONSE');

$app->getDocument()->addScriptOptions(
	'sample-data',
	[
		'icon' => Uri::root(true) . '/media/system/images/ajax-loader.gif'
	]
);
?>
<?php if ($items) : ?>
	<ul id="sample-data-wrapper" class="list-group list-group-flush">
		<?php foreach($items as $i => $item) : ?>
			<?php
				$db    = Factory::getDbo();
				$query = $db->getQuery(true)
					->select($db->quoteName('extension_id'))
					->from($db->quoteName('#__extensions'))
					->where(
						[
							$db->quoteName('folder') . ' = ' . $db->quote('sampledata'),
							$db->quoteName('type') . ' = ' . $db->quote('plugin'),
							$db->quoteName('element') . ' = ' . $db->quote($item->name),
						]
					);

				$db->setQuery($query);

				$id = $db->loadResult();
				$loadUrl = 'index.php?option=com_plugins&view=plugin&task=plugin.edit&extension_id=' . $id . '&amp;' . Session::getFormToken() . '=1';
			?>
			<li class="list-group-item sampledata-<?php echo $item->name; ?>">
				<div class="d-flex justify-content-between align-items-center">
					<div class="me-2">
						<span class="icon-<?php echo $item->icon; ?>" aria-hidden="true"></span>
						<?php echo htmlspecialchars($item->title, ENT_QUOTES, 'UTF-8'); ?>
					</div>
					<div class="btn-group">
						<button type="button" class="btn btn-secondary btn-sm apply-sample-data" data-type="<?php echo $item->name; ?>" data-steps="<?php echo $item->steps; ?>">
							<span class="icon-upload" aria-hidden="true"></span> <?php echo Text::_('JLIB_INSTALLER_INSTALL'); ?>
							<span class="visually-hidden"><?php echo $item->title; ?></span>
						</button>
						<button type=button" class="btn btn-danger btn-sm manage-sample-data" onclick="location.href='<?php echo Route::_($loadUrl); ?>'">
							<span class="icon-tasks" aria-hidden="true"></span> <?php echo Text::_('JACTION_EDIT'); ?>
							<span class="visually-hidden"><?php echo Text::_('JACTION_EDIT'); ?></span>
						</button>
					</div>
				</div>
				<p class="small mt-1"><?php echo $item->description; ?></p>
				<?php // Progress bar ?>
				<div class="sampledata-progress-<?php echo $item->name; ?> d-none mb-3">
					<div class="progress">
						<div class="progress-bar progress-bar-striped progress-bar-animated" role="progressbar"></div>
					</div>
				</div>
				<?php // Progress messages ?>
				<div class="sampledata-progress-<?php echo $item->name; ?> d-none">
					<ul class="list-unstyled"></ul>
				</div>
			</li>
		<?php endforeach; ?>
	</ul>
<?php endif; ?>
```

**Comment by Bakual** [comment]
I've resolved the conflicts and added some margins to the button.
The border was already removed with a different PR elsewhere.

Looks now like this:
image

**Comment by Bakual** [comment]
Fetching the DB is a no-go in my eyes. Also technically "name" doesn't necessary have to be the name of the plugin. It could be anything. It just happens to be the same name for the core plugins.

So if we want to have a direct button, we need to adjust the overview method so the plugin returns its own ID.

**Comment by particthistle** [comment]
I have tested this item :white_check_mark: successfully on 91c4cac07135a22eb0bbff43065d8ba5a7a2f128<br /><br />@Bakual solution is straight forward and elegant. Only feedback is whether the button string should be "Manage Sample Data Plugins"

@infograf768 though I get what you're trying to allow for there, the buttons indicating "Edit" are misleading IMO... you're not "Editing" the sample data, you're managing the availability of the plugin.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33169">issues.joomla.org/tracker/joomla-cms/33169</a>.</sub>

**Comment by Bakual** [comment]
Regarding the name of the button, I think "Manage Sample Data Plugins" gets to long. Maybe @brianteeman has a better idea for the text (I'm no no native english speaker).

**Comment by ChristineWk** [comment]
Buttons:
You don't have e.g: Install Sample Data, therefore there is no need for: Manage Sample Data :-)
In short:
Install
Manage<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33169">issues.joomla.org/tracker/joomla-cms/33169</a>.</sub>

**Comment by Bakual** [comment]
Good point with the string, thanks! Change it.

image

**Comment by ChristineWk** [comment]
I have tested this item :white_check_mark: successfully on 29274b612eb7a4b90ec85437d703bafc00798fd6<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33169">issues.joomla.org/tracker/joomla-cms/33169</a>.</sub>

File: administrator/modules/mod_sampledata/tmpl/default.php [hunk 2]
```diff
@@ -54,4 +54,8 @@
 			</li>
 		<?php endforeach; ?>
 	</ul>
+	<a href="index.php?option=com_plugins&filter[folder]=sampledata" class="btn btn-secondary btn-sm manage-sample-data float-end me-3 mb-3">
+		<span class="icon-tasks" aria-hidden="true"></span> <?php echo Text::_('MOD_SAMPLEDATA_MANAGE_SAMPLEDATA'); ?>
+		<span class="visually-hidden"><?php echo Text::_('MOD_SAMPLEDATA_MANAGE_SAMPLEDATA'); ?></span>
```
Comment (Quy) [inline administrator/modules/mod_sampledata/tmpl/default.php]:
```suggestion
```

This will be read twice by screenreaders.

Comment (Bakual) [reply administrator/modules/mod_sampledata/tmpl/default.php]: Good catch, I just copied those parts from the install button without thinking much about it. Removed it.

**Comment by particthistle** [comment]
I have tested this item :white_check_mark: successfully on ab2019e6e2848e2e5e760192c916112889cb3103<br /><br />Tested after code clean up and button name change. All works.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33169">issues.joomla.org/tracker/joomla-cms/33169</a>.</sub>

**Comment by ChristineWk** [comment]
I have tested this item :white_check_mark: successfully on ab2019e6e2848e2e5e760192c916112889cb3103<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33169">issues.joomla.org/tracker/joomla-cms/33169</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33169">issues.joomla.org/tracker/joomla-cms/33169</a>.</sub>

**Pull request merged**